sql-plugin/src/main/scala/org/apache/spark/rapids/hybrid/HybridParquetScanRDD.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.rapids.hybrid
+
+import com.nvidia.spark.rapids.{CoalesceSizeGoal, GpuMetric}
+
+import org.apache.spark.{InterruptibleIterator, Partition, TaskContext}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.execution.metric.SQLMetric
+import org.apache.spark.sql.types.{StructField, StructType}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+class HybridParquetScanRDD(scanRDD: RDD[ColumnarBatch],
+                           outputAttr: Seq[Attribute],
+                           outputSchema: StructType,
+                           coalesceGoal: CoalesceSizeGoal,
+                           metrics: Map[String, GpuMetric],
+                         ) extends RDD[InternalRow](scanRDD.sparkContext, Nil) {
+
+  private val hybridScanTime = GpuMetric.unwrap(metrics("HybridScanTime"))
+
+  override protected def getPartitions: Array[Partition] = scanRDD.partitions
+
+  override def compute(split: Partition, context: TaskContext): Iterator[InternalRow] = {
+    // the wrapping Iterator for the underlying VeloxScan task
+    val metricsIter = new HybridScanMetricsIter(scanRDD.compute(split, context), hybridScanTime)
+
+    val schema = StructType(outputAttr.map { ar =>
+      StructField(ar.name, ar.dataType, ar.nullable)
+    })
+    require(coalesceGoal.targetSizeBytes <= Int.MaxValue,
```

Comment:
I am a little concerned about this. We just merged in a code change that allows the batch size to go above 2 GiB. When I grep through the code I don't see anywhere else that has a check like this. I am fine if gluten has a 2 GiB limit., but we need to fix up the coalesce goal instead of blowing up. I am fine if this is a follow on issue as this feature is off by default.

Reply:
@sperlingxx I assume cuDF column has 2G limit, but for a batch do we need this check?

Reply:
cudf columns have an int32 index. So they cannot go above 2^31 rows. For a byte column, that is 2 GiB, but a short that is 4 GiB, and int is 8 GiB, a long is 16 GIB, and a DECIMAL_128 is 32 GIB.  CUDF is also working on an extended string format that can go larger than 2 GIB of the payload data. We do not have plans to enable this yet, but want to eventually.

Reply:
Hi @revans2,  I am sorry that this seems to be a meaningless check which I have already forgot why to put it on initially.  IIUC, we can simply remove it and change the IntType to LongType.  However, we have to change the corresponding variable in private repo, in which I filed an issue.